src/data/seq/seq.lean

```diff
@@ -14,10 +14,15 @@ coinductive seq (α : Type u) : Type u
 | cons : α → seq α → seq α
 -/
 
+/--
+The characteristic property of a sequence s: if `s.nth n = none` then also `s.nth (n + 1) = none`.
+-/
+def seq.prop {α : Type u} (s : stream (option α)) : Prop := ∀ {n : ℕ}, s n = none → s (n + 1) = none
```

Comment:
I'm not sure this belongs in the `seq` namespace. (Notice that it doesn't work with projection notation.) What about calling it `stream.is_seq`? Then the `seq` type is more obviously just bundling this predicate, and you can write `f.is_seq`.

Reply:
Sounds good! I changed it.